### ISSUE TITLE
Difficult to select all versions in a major

### ISSUE THREAD

**Issue by jaydenseric** [browserslist/browserslist#488]
If you have this in your `package.json`:

```json
{
  "engines": {
    "node": "^12.0.0 || >= 13.7.0"
  }
}
```

It is hard to replicate that in a browserslist query:

```
Node 12 - 13 and Node < 13, Node >= 13.7.0
```

One gotcha is that browserslist has a different behavior than the npm ecosystem when considering a major number on it's own: `Node 12` in browserslist will select only the newest v12 version, but `12` in `package.json` `engines` or `dependencies` selects every version in that major.

It would be nice if browserslist used the [`semver`](https://www.npmjs.com/package/semver) semantics, allowing the same semver version and range strings that we're used to using in `package.json`, e.g.

```
Node ^12.0.0 || >= 13.7.0
```

Or this, which is the same:

```
Node 12 || >= 13.7.0
```

**Comment by ai** [comment]
Good point. It is breaking changes and I do not have a solution right now.

Let's keep the issue for next major release.
